client/my-sites/stats/stats-upsell/index.tsx

```diff
@@ -0,0 +1,157 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { isEnabled } from '@automattic/calypso-config';
+import { PLAN_PREMIUM } from '@automattic/calypso-products';
+import page from '@automattic/calypso-router';
+import { Gridicon, PlanPrice } from '@automattic/components';
+import { Plans } from '@automattic/data-stores';
+import formatCurrency from '@automattic/format-currency';
+import { Button } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import useCheckPlanAvailabilityForPurchase from 'calypso/my-sites/plans-features-main/hooks/use-check-plan-availability-for-purchase';
+import { useSelector } from 'calypso/state';
+import { getSiteOption } from 'calypso/state/sites/selectors';
+import { getUpsellModalStatType } from 'calypso/state/stats/paid-stats-upsell/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+export default function StatsUpsell( { siteId }: { siteId: number } ) {
+	const translate = useTranslate();
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( getSelectedSiteSlug );
+	const plans = Plans.usePlans( { coupon: undefined } );
+	const plan = plans?.data?.[ PLAN_PREMIUM ];
```

Comment:
I think we should probably upsell to personal here (see p1732207672568679/1732192356.916679-slack-C07H21B2W59). We haven't got decision yet but maybe we would add an option to upsell to Premium for Personal plan users.

Reply:
Will tackle this separately, this is copy pasted from the existing modal which will also need swapping to personal / and or handling both personal and premium upsell scenarios.